src/argilla/client/feedback/training/schemas.py

```diff
@@ -381,6 +392,28 @@ def for_chat_completion(
     ) -> "TrainingTaskForChatCompletion":
         return TrainingTaskForChatCompletion(formatting_func=formatting_func)
 
+    @classmethod
+    def for_question_answering(
+        cls,
+        formatting_func: Optional[Callable[[Dict[str, Any]], Union[None, str, List[str], Iterator[str]]]] = None,
+        question: Optional[TextField] = None,
+        context: Optional[TextField] = None,
+        answer: Optional[TextQuestion] = None,
+    ) -> "TrainingTaskForQuestionAnswering":
+        if (question and context and answer) and formatting_func is not None:
+            raise ValueError(
+                "You must provide either `question`, `context` and `answer`, or a `formatting_func`, not both."
+            )
+
+        if formatting_func is not None:
+            if question or context or answer:
+                raise ValueError(
+                    "`formatting_func` is already defined, so you cannot define `question`, `context` and `answer`."
+                )
```

Comment:
We can merge these Exceptions with `if (question or context or answer) and formatting_func is not None`.

Reply:
Correct.